composer.json

```diff
@@ -7,26 +7,29 @@
         "drush/drush": "^10.3",
         "drupal/purge": "^3.0@beta",
         "drupal/purge_queuer_url": "^1.0@RC",
-        "drupal/redis": "^1.0@RC",
-        "drupal/section_purge": "2.3-rc2",
+        "drupal/section_purger": "dev-updates_to_compatible_with_drupal_9",
```

Comment:
Why is the patch being applied this way? Rather than the usual approach of adding a new element to the `.extra.patches` array?

Reply:
Hey @nicksantamaria , thanks for the CR. 
we are doing this because of the compatibility issue. 
this PR works for D9 and D8.9, but current `section_purger` only supports D8, which means `drupal/section_purger` without composer alias cannot install in D9 site. we add `.extra.patches` will never get installed.

more details: https://www.drupal.org/docs/develop/git/using-git-to-contribute-to-drupal/creating-issue-forks-and-merge-requests#s-useissue-forks-to-makecompatibility-fixes-work-with-composer

Reply:
Ah OK thanks for the explainer @vincent-gao!